qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -215,6 +213,46 @@ def _on_config_changed(self, option):
             self.maybe_hide()
         elif option == 'statusbar.padding':
             self._set_hbox_padding()
+        elif option == 'statusbar.widgets':
+            self._draw_widgets()
+
+    def _draw_widgets(self):
+        """Draw statusbar widgets"""
+        # Start with widgets hidden and show them when needed
+        for widget in [self.percentage, self.backforward, self.tabindex]:
+            self._hbox.removeWidget(widget)
+            widget.hide()
+        # Remove previous text widgets if any
+        for text in self.text_widgets:
+            self._hbox.removeWidget(text)
+            text.hide()
+            text.deleteLater()
+        self.text_widgets = []
+
+        if not config.val.statusbar.widgets:
+            return
+
+        # Read the format string and set widgets accordingly
+        for segment in re.split('{|}', config.val.statusbar.widgets):
+            if segment == 'scroll':
```

Comment:
If we are going to keep the format_option style of config, we should probably just  have a simple `{scroll}` number, and add the `[`, `]`, and `%` (and padding) in the format string. That way, people can have <30!> if they wanted instead of `[30%]`.

In order to do this though, we would probably need to remove `statusbar.padding` though...

Reply:
That's what I was thinking when I decided to implement a format option.
I will try to remove the padding, now that I know about it.

Reply:
Ok, there's was no need to remove `statusbar.padding`, setting `self._hbox.setSpacing(0)` did it.